lib/modules/platform/scm-manager/index.ts

```diff
@@ -0,0 +1,338 @@
+    throw new Error('SCM-Manager endpoint not configured');
+  }
+
+  if (!token) {
+    throw new Error('SCM-Manager API token not configured');
+  }
+
+  setBaseUrl(endpoint);
+  setToken(token);
+
+  try {
+    const me = await getCurrentUser();
+    const gitAuthor = `${me.displayName} <${me.mail}>`;
+    const result = { endpoint, gitAuthor };
+
+    logger.debug({ result }, 'Plattform result');
+
+    return result;
+  } catch (err) {
+    logger.debug(
+      { err },
+      'Error authenticating with SCM-Manager. Check your token',
+    );
+    throw new Error('Init: Authentication failure');
+  }
+}
+
+export async function initRepo({
+  repository,
+  gitUrl,
+}: RepoParams): Promise<RepoResult> {
+  const repo = await getRepo(repository);
+  const defaultBranch = await getDefaultBranch(repo);
+  const url = getRepoUrl(repo, gitUrl, getBaseUrl());
+
+  config = {} as any;
+  config.repository = repository;
+  config.defaultBranch = defaultBranch;
+
+  await git.initRepo({
+    ...config,
+    url,
+  });
+
+  // Reset cached resources
+  invalidatePrCache();
+
+  const result = {
+    defaultBranch: config.defaultBranch,
+    isFork: false,
+    repoFingerprint: repoFingerprint(config.repository, getBaseUrl()),
+  };
+
+  logger.trace({ result }, `Repo initialized`);
+
+  return result;
+}
+
+export async function getRepos(): Promise<string[]> {
+  const repos = (await getAllRepos()).filter((repo) => repo.type === 'git');
+  return repos.map((repo) => `${repo.namespace}/${repo.name}`);
+}
+
+export async function getBranchPr(branchName: string): Promise<Pr | null> {
+  return await findPr({ branchName, state: 'open' });
+}
+
+export async function findPr({
+  branchName,
+  prTitle,
+  state = 'all',
+}: FindPRConfig): Promise<Pr | null> {
+  const inProgressPrs = await getPrList();
+  const result = inProgressPrs.find(
+    (pr) =>
+      branchName === pr.sourceBranch &&
+      (!prTitle || prTitle === pr.title) &&
+      matchPrState(pr, state),
+  );
+
+  if (result) {
+    logger.trace({ result }, `Found PR`);
+    return result;
+  }
+
+  logger.trace(
+    `Could not find PR with source branch ${branchName} and title ${
+      prTitle ?? ''
+    } and state ${state}`,
+  );
+
+  return null;
+}
+
+export async function getPr(number: number): Promise<Pr | null> {
+  const inProgressPrs = await getPrList();
+  const cachedPr = inProgressPrs.find((pr) => pr.number === number);
+
+  if (cachedPr) {
+    logger.trace('Returning from cached PRs');
+    return cachedPr;
+  }
+
+  try {
+    const result = await getRepoPr(config.repository, number);
+    logger.trace('Returning PR from API');
+    return mapPrFromScmToRenovate(result);
+  } catch (error) {
+    logger.error({ error }, `Can not find a PR with id ${number}`);
+    return null;
+  }
+}
+
+export async function getPrList(): Promise<Pr[]> {
+  if (config.prList === null) {
+    try {
+      config.prList = (await getAllRepoPrs(config.repository)).map((pr) =>
```

Comment:
note: no cache yet, so can be expensive when there are a lot pr's

Reply:
The PR is big as is so this can be done in a new PR.